src/components/ContinuationIn/ExtraproRegistration.vue

```diff
@@ -237,58 +323,104 @@ export default class ExtraproRegistration extends Vue {
-    this.business = this.getContinuationInBusinessInfo || { ...EmptyBusinessLookup }
+    // set existing business info object from the store, if it exists
+    if (this.getExistingBusinessInfo) this.business = this.getExistingBusinessInfo
 
     // if mode is LOOKUP, set this component to active (which hides the other component)
     if (this.business.mode === 'LOOKUP') this.active = true
   }
 
-  setBusiness (business: BusinessLookupResultIF) {
-    this.business = { ...business } // for reactivity
-    this.business.mode = 'LOOKUP'
-    this.setContinuationInBusinessInfo(this.business)
+  async setBusiness (result: BusinessLookupResultIF): Promise<void> {
+    // FUTURE: fetch extra business data from COLIN and add to business object
+    // NOTE: fetch for both active and historical businesses
+    // const businessInfo = await LegalServices.fetchColinBusinessInfo(business.identifier)
+    //   .catch(() => {})
+
+    // FUTURE: convert FD to FEDERAL and add other COLIN conversions here
+
+    this.business = {
+      homeJurisdiction: null, // get from COLIN
+      businessIdentifier: result.identifier,
+      businessLegalName: result.name,
+      identifier: null, // get from COLIN
+      incorporationDate: null, // get from COLIN
+      legalName: null, // get from COLIN
+      mode: 'LOOKUP',
+      status: result.status,
+      taxId: result.bn?.substring(0, 9) || null
+    }
+    this.setExistingBusinessInfo(this.business)
     this.active = true
   }
 
   /** Resets this component back to its initial state. */
   reset () {
-    this.business = { ...EmptyBusinessLookup }
-    this.setContinuationInBusinessInfo(this.business)
+    this.business = {} as unknown as ExistingBusinessInfoIF
+    this.setExistingBusinessInfo(this.business)
     // set this component to inactive (which shows the other component)
     this.active = false
-    this.checkbox = false
   }
 
-  // *** TODO: add watchers for jurisdiction and affidavit
-  @Watch('business')
-  @Watch('checkbox')
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async fileSelected (file): Promise<void> {
+    // *** TODO: finish implementing -- see UploadMemorandum.vue
+    if (file) {
+      // add property reactively to business object
+      this.$set(this.business, 'affidavitFileKey', '123')
+    } else {
+      // delete property reactively if the file is cleared
+      this.$delete(this.business, 'affidavitFileKey')
+      // FUTURE: should also delete the file from Minio
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  isFileUploadValidFn (val): void {
+    // *** TODO: implement -- see UploadMemorandum.vue
+  }
+
+  @Watch('business.isUlc')
+  onIsUlcChanged (val: boolean): void {
+    // delete the affidavit property if the business is not a ULC
+    if (!val) {
+      delete this.business.affidavitFileKey
+    }
+  }
+
+  @Watch('business', { deep: true })
```

Comment:
Out of curiosity, why are you using deep here Sev? Do you really need it in this case?

Reply:
Because I want to watch for the properties in the object changing. The object itself isn't changing.

Reply:
For the record, I would not need the deep option here if I kept changing the business object reference (by assigning a new object each time), eg,

`this.business = { ...this.business, myParam: 123 }`